config/main.py

```diff
@@ -1094,11 +1122,19 @@ def add(ctx, interface_name, ip_addr):
     try:
         ipaddress.ip_network(unicode(ip_addr), strict=False)
         if interface_name.startswith("Ethernet"):
-            config_db.set_entry("INTERFACE", (interface_name, ip_addr), {"NULL": "NULL"})
-            config_db.set_entry("INTERFACE", interface_name, {"NULL": "NULL"})
+            if VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
+                config_db.set_entry("VLAN_SUB_INTERFACE", interface_name, {"admin_status": "up"})
```

Comment:
May be it is good now to just set a `table_name` here and use `config_db.set_entry` once instead of repeated lines of same code.